### PR TITLE
[JENKINS-60005, JENKINS-61862] - Update for Windows Service Wrapper 2.8.0

### PR DESCRIPTION
**Pull request by NextTurn**
Supported by #42.

- Removes *jenkins-slave.exe.config*.
  - Uses the .NET FX 4.0 binary. [JENKINS-61862](https://issues.jenkins-ci.org/browse/JENKINS-61862)
  - Removes `DontEnableSchUseStrongCrypto=false`. winsw/winsw#498
- Removes `ERROR_ELEVATION_REQUIRED` handling. winsw/winsw#457

### CODE REVIEW

**Comment by NextTurn** [comment]
Not sure if the build failure is relevant.

**Comment by oleg-nenashev** [comment]
https://github.com/jenkins-infra/jenkins.io/pull/3295 documents the policy so that we are able to move to .NET Framework 4.0

**Comment by NextTurn** [comment]
Supported by #42.

File: src/main/java/org/jenkinsci/modules/windows_slave_installer/WindowsSlaveInstaller.java [hunk 1]
```diff
@@ -124,9 +90,6 @@ public void install(LaunchConfiguration params, Prompter prompter) throws Instal
         final File agentExe = new File(dir, "jenkins-slave.exe");
         FileUtils.copyURLToFile(WindowsSlaveInstaller.class.getResource("jenkins-slave.exe"), agentExe);
 
-        FileUtils.copyURLToFile(WindowsSlaveInstaller.class.getResource("jenkins-slave.exe.config"),
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/modules/windows_slave_installer/WindowsSlaveInstaller.java:127]: Should it also delete the existing one if it exists? AFAICT there should be no harm in the default configuration except non-surprising crashes on .NET Framework 2.0, but there should be no other risks. At the same time, there is a risk that user instances override the file.

File: src/main/java/org/jenkinsci/modules/windows_slave_installer/WindowsSlaveInstaller.java [hunk 2]
```diff
@@ -72,35 +66,7 @@ public Localizable getConfirmationText() {
...
-            return new LocalLauncher(out).launch().cmds(agentExe, command).stdout(out).pwd(pwd).join();
-        } catch (IOException e) {
-            if (e.getMessage().contains("CreateProcess") && e.getMessage().contains("=740")) {
-                // fall through
-            } else {
-                throw e;
-            }
-        }
-
-        // error code 740 is ERROR_ELEVATION_REQUIRED, indicating that
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/modules/windows_slave_installer/WindowsSlaveInstaller.java:85]: IIRC this code did not quite work on Windows 10 anyway. Do not recall details. Switching to a built-in elevation logic is a right step anyway

File: src/main/java/org/jenkinsci/modules/windows_slave_installer/WindowsSlaveInstaller.java (hunk 1 above)
Comment (NextTurn) [reply src/main/java/org/jenkinsci/modules/windows_slave_installer/WindowsSlaveInstaller.java:127]: Agreed.

**Comment by NextTurn** [comment]
CI is very unstable... I've seen a number of failures like this:

```
[2020-06-23T08:28:08.709Z] Recording test results
[2020-06-23T08:28:08.881Z] No test report files were found. Configuration error?
No test report files were found. Configuration error?
```

**Comment by oleg-nenashev** [comment]
> [2020-06-23T08:28:08.881Z] No test report files were found. Configuration error?
No test report files were found. Configuration error?

it looks to be a red herring. One parallel execution fails the SpotBugs check, and it interrupts other builds which then fail with "No tests". It is a regression after https://github.com/jenkins-infra/pipeline-library/pull/88 in the Pipeline library, CC @darxriggs . IMO we should stop failing individual parallel stages when there is no tests, and fail the build later if there is no tests overall. Anyway, this is a potential improvement to UX, not a critical issue in the Pipeline implementation

**Comment by NextTurn** [comment]
> One parallel execution fails the SpotBugs check, and it interrupts other builds which then fail

In that case it could be cheaper to mark other builds as "canceled".

**Comment by oleg-nenashev** [comment]
> In that case it could be cheaper to mark other builds as "canceled".

It is basically what Jenkins does in such case. At the same time the Pipeline library uses try/finally, so it tries to publish JUnit even if the execution is interrupted. Obviously, we should replace finally with `catch` which ignores Interruptions

```
                                try {
                                    infra.runWithJava(command, jdk, null, addToolEnv)
                                } finally {
                                    if (!skipTests) {
                                        junit('**/build/test-results/**/*.xml')
                                    }
                                }
```

**Comment by NextTurn** [comment]
It seemed to be a connection issue this time.

> [2020-06-23T09:32:54.843Z] Cannot contact EC2 (aws) - Windows 2019 (i-08462416792308337): hudson.remoting.ChannelClosedException: Channel "hudson.remoting.Channel@3e754efa:EC2 (aws) - Windows 2019 (i-08462416792308337)": Remote call on EC2 (aws) - Windows 2019 (i-08462416792308337) failed. The channel is closing down or has closed down

**Comment by oleg-nenashev** [comment]
CI is back to normal. We still experience issues with AWS after moving some of our agents there from Azure (costs and sponsorship, no technical reason for multi-cloud)

**Pull request merged**